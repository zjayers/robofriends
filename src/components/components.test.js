import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import App from '../containers/App'
import Card from './Card';
import CardList from './CardList';
import CounterButton from './CounterButton';
import ErrorBoundary from './ErrorBoundry';
import Header from './Header';
import Scroll from './Scroll';
import SearchBox from './SearchBox';

it('expect functional components to match snapshots', () => {
  expect(toJson(shallow(<Card />))).toMatchSnapshot();

  const mockRobots = [
    {
      id: 1,
      name: 'Rob',
      userName: 'Rob',
      email: 'Rob@rob.com'
    }
  ]

  expect(toJson(shallow(<CardList robots={mockRobots} />))).toMatchSnapshot();
})

it('expect class based components to match snapshots', () => {

  const mockColor = 'red';
  expect(toJson(shallow(<CounterButton color={mockColor} />))).toMatchSnapshot();

})

it ('class button correctly increments the counter', () => {
  const mockColor = 'red';
  const wrapper = shallow(<CounterButton color={mockColor} />)

  wrapper.find('[id="counter"]').simulate('click');
  expect(wrapper.state()).toEqual({count: 2});

})
