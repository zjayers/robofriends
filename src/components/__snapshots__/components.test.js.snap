// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect class based components to match snapshots 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  1
</button>
`;

exports[`expect functional components to match snapshots 1`] = `
<div
  className="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
>
  <img
    alt="robots"
    src="https://robohash.org/undefined?size=200x200"
  />
  <div>
    <h2 />
    <p />
  </div>
</div>
`;

exports[`expect functional components to match snapshots 2`] = `
<div>
  <Card
    email="Rob@rob.com"
    id={1}
    key="0"
    name="Rob"
  />
</div>
`;
